microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/MultiAppHealthTest.java

```diff
@@ -0,0 +1,82 @@
+package org.jboss.eap.qe.microprofile.health;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+
+@RunWith(Arquillian.class)
+public abstract class MultiAppHealthTest {
+
+    @ContainerResource
+    ManagementClient managementClient;
+
+    String healthURL;
+
+    @Before
+    public void composeHealthEndpointURL() {
+        healthURL = "http://" + managementClient.getMgmtAddress() + ":" + managementClient.getMgmtPort() + "/health";
+    }
+
+    /**
+     * @tpTestDetails Customer multi-deployment / deployment with sub-deployment scenario.
+     * @tpPassCrit Overall and the health check status is up.
+     * @tpSince EAP 7.3.0.CD19
+     */
+    @Test
+    @RunAsClient
+    public void testHealthEndpoint() {
+        RestAssured.when().get(this.healthURL).then()
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("UP"),
+                        "checks", hasSize(3), // BothHealthCheck contains both annotations: @Liveness and @Readiness
```

Comment:
Yes, I see your point. The abstract class tests expect sub-classes to deploy something specific / limited. What about having a sort of TestDelegation class instead of this abstract class?

Reply:
> The abstract class tests expect sub-classes to deploy something specific / limited.

Correct. The key point of my consideration is to not scatter implementation logic over hierarchy. 
Any "pattern" that would solve this is good IMO. Delegation to run some supplied code would work, I guess, but then what's the point for that "empty" superclass? 
I - for instance - would start from removing the "details-aware" code from the superclass to subclasses - i.e. I'd move the checks down, but that would just be my immediate coding attempt. 
Then the design could be adjusted further, somebody else could choose composition. 
I'd say there are several solutions and you know better the detail.

Reply:
Yeah, I would get rid of the hierarchy....Fixing...